File: rating-service/src/main/java/com/example/rating_service/repo/RatingRepo.java
package com.example.rating_service.repo;

import com.example.rating_service.model.entity.RatingEntity;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface RatingRepo extends MongoRepository<RatingEntity, String> {
    Optional<RatingEntity> findByTargetUserId(UUID targetUserId);
}

File: rating-service/src/main/kotlin/com/example/rating_service/repo/RatingRepo.kt
package com.example.rating_service.repo

import com.example.rating_service.entity.RatingEntity
import java.util.UUID
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface RatingRepo : MongoRepository<RatingEntity, UUID> {
    fun findByTargetUserId(targetUserId: UUID): List<RatingEntity>

    fun existsByRideId(rideId: UUID): Boolean
}